src/main/java/com/salesforce/dynamodbv2/mt/util/CachingAmazonDynamoDbStreams.java

```diff
@@ -487,6 +512,13 @@ private static String toShortString(GetRecordsResult result) {
     // iterator cache
     private final LoadingCache<CachingShardIterator, String> iteratorCache;
 
+    private Cache<String, DescribeStreamResult> describeStreamCache;
+
+    @VisibleForTesting
+    public Cache<String, DescribeStreamResult> getDescribeStreamCache() {
+        return describeStreamCache;
+    }
```

Comment:
As discussed, I'd prefer to not expose the internal cache. Some possible strategies for testing could be:
- introduce builder/constructor variant that allows passing in cache reference
- assert cache hits/misses by observing calls to the passed in AmazonDynamoDBStreams instance

We can address this in a separate PR if we think it's worth doing.